#117 associate a student with their modules via side bar

PhilippeLeopoldie opened the pull request:
This PR enhances the Sidebar and related components with user context, course-based navigation, and improved authorization.

Changes

- Sidebar.razor:
   - Added required using directives and injected services.
   - Updated AuthorizeView to use Context for authentication context access for student.
   - Navigation items now dynamically reference CourseId.
   - Added new navigation items and a @code block to manage user state and course ID resolution.

- ModuleMenu
   - Wrapped menu contents in <AuthorizeView> to restrict visibility to users with the Teacher role.

PhilippeLeopoldie committed:
Enhance Sidebar.razor with user context and navigation

Updated `Sidebar.razor` to include additional using directives and inject necessary services. Modified `AuthorizeView` to utilize a `Context` parameter for authentication context access. Navigation items now dynamically reference `CourseId`, and new items have been added. Introduced a new `@code` block to manage user authentication state and set `CourseId` based on the authenticated user.

PhilippeLeopoldie committed:
Add courseId claim to user claims in AuthService

This update introduces a new claim for the user's course ID in the AuthService.cs file. The claim defaults to 0 if the course ID is null, enhancing the application's ability to manage user claims for authorization and related functionalities.

PhilippeLeopoldie committed:
Merge branch '#106-module-date-overlapping-validation' into #117-associate-a-student-with-their-modules-via-sideBar

PhilippeLeopoldie committed:
Add authorization to ModuleMenu for Teacher role

Wrap the ModuleMenu method's content in an <AuthorizeView> component to restrict access to users with the "Teacher" role. The dropdown menu remains unchanged but is now only visible to authorized users. The "Delete" button has been moved inside the <AuthorizeView> block to ensure proper access control.

PhilippeLeopoldie committed:
Clean up formatting by removing unnecessary line breaks

PhilippeLeopoldie committed:
Remove unused courseId claim

PhilippeLeopoldie committed:
Refactor claim addition in AuthService.cs

Updated the claim for the user's name identifier in the AuthService.cs file. The change involved removing and re-adding the same line, likely for formatting or consistency purposes, with no impact on the logic or functionality.

JorgenHagglund reviewed (approved):
Funkar bra!

ahlgren1234 reviewed (approved):
Looking good

PhilippeLeopoldie committed:
Merge from development branch

ChornaIryna commented on LMS.Blazor/Components/Layout/Sidebar.razor:
```diff
@@ -18,3 +23,15 @@
+        var authUser = authState.User;
+        var userId = authUser.FindFirstValue(ClaimTypes.NameIdentifier);
+        User = await UserManager.FindByIdAsync(userId);
+        CourseId = User.CourseId ?? 0;
```
Nu fick jag felet, men det fanns inte där tidigare i lärarens konto. CourseId = User.CourseId ?? 0; kan leda till att länken blir /courses/0/modules om CourseId saknas.
<img width="1804" height="414" alt="image" src="https://github.com/user-attachments/assets/96cc4ac5-096c-4cb6-82b2-87cde595d1bb" />

PhilippeLeopoldie replied:
Jag kan inte återskapa felet :S. Det verkar komma från Home.razor. Har du några nya ändringar där lokalt?

PhilippeLeopoldie committed:
Add CourseStateService with CourseDto property

Introduces the `CourseStateService` class in the
`LMS.Blazor.Client.Services` namespace. Adds a nullable
`Course` property of type `CourseDto?` and includes
the necessary `using` directive for `CourseDto`.

JorgenHagglund reviewed (approved):
Ser bra ut!

PhilippeLeopoldie committed:
Merge branch 'development' into #117-associate-a-student-with-their-modules-via-sideBar
